anvil-python/anvil/commands/haproxy.py

```diff
@@ -100,34 +142,55 @@ def has_prompts(self) -> bool:
         return True
 
     def prompt(self, console: Console | None = None) -> None:
-        variables = questions.load_answers(
-            self.client, self._KEEPALIVED_CONFIG
-        )
+        variables = questions.load_answers(self.client, self._HAPROXY_CONFIG)
         variables.setdefault("virtual_ip", "")
+        variables.setdefault("charm_haproxy_config", {})
 
         # Set defaults
         self.preseed.setdefault("virtual_ip", "")
+        self.preseed.setdefault("charm_haproxy_config", {})
 
-        keepalived_config_bank = questions.QuestionBank(
-            questions=keepalived_questions(),
+        haproxy_config_bank = questions.QuestionBank(
+            questions=haproxy_questions(),
             console=console,
             preseed=self.preseed.get("haproxy"),
             previous_answers=variables,
             accept_defaults=self.accept_defaults,
         )
 
-        variables["virtual_ip"] = keepalived_config_bank.virtual_ip.ask()
+        variables["virtual_ip"] = haproxy_config_bank.virtual_ip.ask()
+
+        cert_filepath = haproxy_config_bank.ssl_cert.ask()
+        key_filepath = haproxy_config_bank.ssl_key.ask()
+        if cert_filepath is not None and key_filepath is not None:
+            with open(cert_filepath) as cert_file:
+                cert = cert_file.read()
+            with open(key_filepath) as key_file:
+                key = key_file.read()
+            if not os.path.isdir(HAPROXY_CERTS_DIR):
+                mkdir(HAPROXY_CERTS_DIR)
+            with open(
+                os.path.join(HAPROXY_CERTS_DIR, "haproxy.pem"), "w"
+            ) as combined_file:
+                combined_file.write(key + cert)
+            variables["charm_haproxy_config"]["services"] = (
+                self._TLS_SERVICES_CONFIG
+            )
+            variables["haproxy_port"] = 443
```

Comment:
Let's keep only this line but set it in `extra_tfvars` below. It is something that is not coming by prompt but it is set based on prompt answers